src/host.rs

```diff
@@ -192,6 +192,15 @@ impl<'a> HostAndPort<&'a str> {
     }
 }
 
+impl<S: AsRef<str>> fmt::Display for HostAndPort<S> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        try!(self.host.fmt(f));
```

Comment:
@SamWhited I don’t understand what you’re proposing. Could you explain some more or give an example?

In any case, it is incorrect for a `Display::fmt` impl to return `Err` other than for propagating errors that ultimately come from the `Formatter`. Formatting is supposed to be infallible, it’s only writing to the output stream (for example to a file on a full disk) that can fail.